internal/client/client.go

```diff
@@ -0,0 +1,141 @@
+package client
```

Comment:
Most of this file is lifted from `provider.go` untouched, except `Do` which is new, and a bug fix in `loadCertFolder`.